### PR TITLE
common/params: support nonblocking write

### PR DESCRIPTION
**Pull request by deanlee**
continue https://github.com/commaai/openpilot/pull/25912.  fixed the pytest issues by passing `params_prefix` to `AsyncWriter`

### CODE REVIEW

File: common/params.cc [hunk 1]
```diff
@@ -327,3 +334,31 @@ void Params::clearAll(ParamKeyType key_type) {
 
   fsync_dir(getParamPath());
 }
+
+void Params::putNonBlocking(const std::string &key, const std::string &val) {
+  static AsyncWriter async_writer;
```
Comment (adeebshihadeh) [inline common/params.cc:339]: this whole thing seems simpler if the AsyncWriter belongs to the Params instance. I think it's actually desirable that it blocks if you do `Params().putNonBlocking()`. we really shouldn't be repeatedly creating params instances

Comment (deanlee) [reply common/params.cc:339]: I'll try that

**Comment by deanlee** [comment]
@adeebshihadeh The disadvantage of moving AsyncWriter to Params is that each params has an instance of future and queue. the static version is a little more complicated, but it doesn't have this problem. What do you think?

**Comment by adeebshihadeh** [comment]
@deanlee is this ready to review?

**Comment by deanlee** [comment]
@adeebshihadeh  reday for review

**Pull request merged**